server/src/main/java/io/crate/lucene/FunctionToQuery.java

```diff
@@ -21,20 +21,77 @@
 
 package io.crate.lucene;
 
-import io.crate.expression.symbol.Function;
-import io.crate.metadata.Scalar;
+import java.util.List;
+
+import javax.annotation.Nullable;
 
 import org.apache.lucene.search.Query;
 
-import javax.annotation.Nullable;
-import java.io.IOException;
+import io.crate.expression.symbol.Function;
+import io.crate.expression.symbol.Literal;
+import io.crate.expression.symbol.Symbol;
+import io.crate.lucene.LuceneQueryBuilder.Context;
+import io.crate.metadata.Reference;
 
-/**
- * @deprecated Implement {@link Scalar#toQuery(Function, io.crate.lucene.LuceneQueryBuilder.Context)} instead.
- **/
-@Deprecated
 public interface FunctionToQuery {
 
+    /**
+     * Returns a Lucene Query with the semantics of the Function.
+     *
+     * <p>
+     * Called if the function is used in the `WHERE` part of a statement when
+     * selecting data from a Table with a Lucene index.
+     * </p>
+     *
+     * <p>
+     * If `null` is returned, a fallback implementation is used that loads
+     * individual records and evaluates the scalar function on each row.
+     * </p>
+     *
+     * <p>
+     * Default implementation calls {@link #toQuery(Reference, Literal, Context)},
+     * which is there to make it more convenient to implement the functionality.
+     * It unwraps the arguments for the common `col <op> literal` case.
+     * </p>
+     **/
     @Nullable
-    Query apply(Function input, LuceneQueryBuilder.Context context) throws IOException;
+    default Query toQuery(Function function, LuceneQueryBuilder.Context context) {
+        List<Symbol> arguments = function.arguments();
+        if (arguments.size() == 2
+                && arguments.get(0) instanceof Reference ref
+                && arguments.get(1) instanceof Literal<?> literal) {
+            return toQuery(ref, literal, context);
+        }
+        return null;
+    }
+
+    /**
+     * See {@link #toQuery(Function, Context)}
+     **/
+    @Nullable
+    default Query toQuery(Reference ref, Literal<?> literal, Context context) {
```

Comment:
Personally I am not so much of a fan of default implementations in interfaces returning null. However, i guess the alternative here would be to make 3 interfaces for each to `toQuery` variant and only use the one in the implementations which are relevant which seems overkill. So I guess this makes sense.

Reply:
Or having a lot of return null methods in the implementations.

Reply:
I agree that it is a bit strange, but considering the trade-offs I think it is the best option.